### DISCUSSION TITLE
How do I deploy the OEP for production?

### DISCUSSION THREAD

**Discussion by pajot** [openego/oeplatform#882] [Q&A] [answered]
Hi everyone,

the installation tutorial only gets me as far as installing for testing purposes. How is the OEP deployed for production, say, using wsgi and a webserver such as nginx?

Thanks

Stephen

**Comment by wingechr** [comment, marked as answer]
Hi Stephen.

the oep uses the python django framework. There should be a ton of tutorials how to set up django for your web server of choice.

In general: 
* point your webserver to the main wsgi script (`oeplatform/wsgi.py`)
* make sure that the virtual environment is also loaded.
* make sure the static files are served correctly

This is how I would set it up with apache2, (I am not too good with nginx):

```conf

# main wsgi script
<Directory OEP_INSTALL_DIR/oeplatform>
        <Files wsgi.py>
                Require all granted
        </Files>
</Directory>

# static files
Alias /static/ OEP_INSTALL_DIR/static/
<Directory  OEP_INSTALL_DIR/static>
        Require all granted
</Directory>

# wsgi process group / virtualenv
WSGIDaemonProcess oep processes=3 threads=10 user=www-data display-name=oep python-path=OEP_INSTALL_DIR:OEP_INSTALL_DIR/.env/lib/PYTHON_VERSION/site-packages
WSGIApplicationGroup %{GLOBAL}
WSGIScriptAlias / OEP_INSTALL_DIR/wsgi.py process-group=oep

# pass on token header
WSGIPassAuthorization On
```

Additionally, you want to configure SSL if the server is publicly visible.


Hope that helps,

Cheers

**Reply by pajot** [reply]
Thanks for this, it's helpful.

Are you using apache2 in your production installation?

I have managed to get our testbed running using nginx, although I'm finding myself having to write start-stop scripts to manage boot time execution.

**Reply by wingechr** [reply]
> Are you using apache2 in your production installation?

I think so (I am not managing the productive environment, but I use apache2 for my own test environment)
